pkg/plugin/validator.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package plugin provides the implementation of the JVS plugin interface.
+package plugin
+
+import (
+	"context"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/abcxyz/pkg/githubapp"
+	"github.com/google/go-github/v55/github"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+)
+
+type Validator struct {
+	cfg        *PluginConfig
+	decodedPem *rsa.PrivateKey
+	client     *github.Client
```

Comment:
github.Client is using a` http.client` for handling the requests. github graqhql can also be queird using `http.Client`, we can just reuse this for handling the email query. 
Currently I think it's a bit unnecessary because we are not querying the emails, and using SSO query requires org admin permission.

Reply:
Yes, but I checked GitHub's graphsql api, we can just use a http client to send out the request.

Reply:
If we really want to use graphQL's client, I can make the change, but I just dont think it's necessary as of now

Reply:
I am fine with it for now.

Reply:
Same here